.github/workflows/test.yml

```diff
@@ -2,32 +2,50 @@ name: Test
 
 on:
   push:
-    branches: ['*']
+    branches: [main]
   pull_request:
-    branches: ['*']
+    branches: [main]
 
 jobs:
   bud:
     name: Test
     runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest]
+        go: ['1.16', '1.17', '1.18']
+        node: ['16']
+        include:
+          - os: ubuntu-latest
+            go: 1.18
+            node: 14
+          - os: ubuntu-latest
+            go: 1.18
+            node: 18
+          - os: macos-latest
+            go: 1.18
+            node: 14
+          - os: macos-latest
+            go: 1.18
+            node: 18
```

Comment:
only do the full Node.js matrix testing for the latest Go version

Reply:
Looks good, thanks!